services/graph/pkg/service/v0/drives.go

```diff
@@ -408,7 +409,21 @@ func (g Graph) UpdateDrive(w http.ResponseWriter, r *http.Request) {
 
 	if drive.Quota.HasTotal() {
 		user := revactx.ContextMustGetUser(r.Context())
-		canSetSpaceQuota, err := g.canSetSpaceQuota(r.Context(), user)
+
+		// NOTE: a space admin cannot get a space by ID. We need to fetch all spaces and search for it
+		dt := _spaceTypePersonal
+		filters := []*storageprovider.ListStorageSpacesRequest_Filter{listStorageSpacesTypeFilter(_spaceTypeProject)}
+		res, err := g.ListStorageSpacesWithFilters(r.Context(), filters, true)
+		if err == nil && res.GetStatus().GetCode() == rpc.Code_CODE_OK {
```

Comment:
no own error if `res.GetStatus().GetCode() != rpc.Code_CODE_OK`?

Reply:
No. This is only to catch the error early. The storageprovider will check the type and permission later again. I would like to remove this completely. But pipeline it is not very happy if I do so.

Reply:
nice, thanks for explaining